### PR TITLE
Class map tag override

### PR DESCRIPTION
**Pull request by raithedavion**
## Description

Please provide a clear and concise description of your changes.
Explain what your PR does and why it's necessary.

I had a use case for using this in conjunction with a WYSIWYG editor, and I needed specific css classes that correspond to the editor to be mapped to specific styles in a template word document.  So I made the required changes to the module and so far my local changes work very well.  

## Issue Reference

Link to the issue this PR addresses, if applicable.
Example: Fixes #123

## Checklist Before Requesting a Review

- [x] I have performed a self-review of my code.
- [x] My code follows the project's coding style and guidelines.
- [x] I have run tests and verified that all existing and new tests pass.
- [x] I have added new tests to cover my changes.

### CODE REVIEW

File: html4docx/h4d.py [hunk 1]
```diff
@@ -46,53 +60,244 @@ def set_initial_attrs(self, document = None):
...
+            paragraph.style = style_name
+            return True
+        except KeyError:
+            # Style doesn't exist in document
+            print(
+                f"Warning: Style '{style_name}' not found in document. Using default."
+            )
+            return False
+
+    def apply_style_to_run(self, run, style_name):
```
Comment (dfop02) [inline html4docx/h4d.py]: Since we always call it using `self.run` for different styles, I think we could remove it from params and use `self.run` directly on method.

**Comment by raithedavion** [comment]
Requested changes made.  Let me know if you want me to pull down your current "HISTORY" or if you want to manually merge that file.

File: tests/test_h4d.py [hunk 2]
```diff
@@ -1155,43 +1260,448 @@ def test_nested_styles_on_multiple_tags(self):
...
+        self.assertEqual(doc.paragraphs[0].style.name, "Normal")
+
+    def test_none_style_map(self):
+        """Test with None style_map"""
+        self.document.add_heading("Test: Test None as style map", level=1)
+        html = "<p>Test</p>"
+
+        doc = Document()
+        parser = HtmlToDocx(style_map=None)
+        parser.options["style-map"] = True
```
Comment (dfop02) [inline tests/test_h4d.py]: Since these flags will be set to `True` by default, we should remove all of these lines from `test_h4d.py`, unless we're explicitly testing the `False` case.

File: README.md [hunk 3]
```diff
@@ -98,11 +99,14 @@ All table styles we support can be found [here](https://python-docx.readthedocs.
 
 #### Options
 
-There is 4 options that you can use to personalize your execution:
+There is 6 options that you can use to personalize your execution:
+
 - Disable Images: Ignore all images.
 - Disable Tables: If you do it, it will render just the raw tables content
 - Disable Styles: Ignore all CSS styles.
```
Comment (dfop02) [inline README.md]: @raithedavion Please make sure that when the `parser.options['styles']` option is disabled, `parser.options['style-map']` is also disabled, since it’s treated as a CSS-style configuration. However, `parser.options['tag-override']` feels more like a DOCX-style setting rather than CSS, so I believe we can keep it enabled. What do you think?

File: tests/test_h4d.py [hunk 4]
```diff
@@ -1155,43 +1260,448 @@ def test_nested_styles_on_multiple_tags(self):
...
+
+        # Should not raise exception
+        try:
+            parser.add_html_to_document(html, self.document)
+            success = True
+        except Exception:
+            success = False
+
+        self.assertTrue(success)
```
Comment (dfop02) [inline tests/test_h4d.py]: I think you could include here if the style fallback to the default as expected. Not sure if you already tested.

File: tests/test_h4d.py [hunk 5]
```diff
@@ -1155,43 +1260,448 @@ def test_nested_styles_on_multiple_tags(self):
...
+        html = '<p><span style="font-style: italic">Italic text</span></p>'
+        self.document.add_heading("Test: Test inline italics", level=1)
+
+        doc = Document()
+        parser = HtmlToDocx()
+        parser.add_html_to_document(html, self.document)
+        parser.add_html_to_document(html, doc)
+
+        run = doc.paragraphs[0].runs[0]
+        self.assertTrue(run.font.italic)
```
Comment (dfop02) [inline tests/test_h4d.py]: We already have a similar test in “Test: Bold, Italic, Underline, and Strike tags”, but it doesn’t cover the new CSS properties you added. Here, the focus should be on testing `font-weight`, `font-style`, `text-decoration`, and `font-family`. Please create a test case for each of them, testing a few scenarios, also include tests for receiving `none` as value.
Good job!

**Comment by raithedavion** [comment]
Planning on getting it this weekend.  Unfortunately had other priorities this week :D

**Comment by raithedavion** [comment]
Well, I am having a heck of a time rebasing it and pulling the changes that got published.  I'm going to try and re-create my local branch and go from there.  But working on it :D

**Comment by dfop02** [comment]
Take your time. You might even be able to reuse some of the new features, so hopefully it won’t take too long 😄
Once you’re done, just let me know!

File: html4docx/h4d.py [hunk 6]
```diff
@@ -86,6 +107,196 @@ def copy_settings_from(self, other):
...
+                clean_value = re.sub(
+                    r"\s*!important\s*", "", value, flags=re.IGNORECASE
+                ).strip()
+                important_styles[prop] = clean_value
+            else:
+                normal_styles[prop] = value
+
+        return normal_styles, important_styles
+
+    def apply_inline_styles_to_run(self, styles_dict):
```
Comment (dfop02) [inline html4docx/h4d.py]: I believe this entire method is already covered in the latest release, including the newly supported CSS styles. Since this function is declared but never used, I believe we can safely remove it and rely on the existing priority order logic for **important**, **overwrite**, and **map**.

File: HISTORY.rst [hunk 7]
```diff
@@ -4,22 +4,23 @@ Release History
...
 
-- None
-
+- Added support for custom css class to word style mappings
+- Added support for html tag to style overrides
+- Added support for setting default word style for new documents.
+- Added support for "!important" style precedence.   
 
-1.1.1 (2025-11-26)
+1.1.1 (2025-11-xx)
```
Comment (dfop02) [inline HISTORY.rst]: Keep this one as 2025-11-26, please.

File: html4docx/h4d.py [hunk 8]
```diff
@@ -1326,6 +1497,25 @@ def handle_starttag(self, tag, attrs):
...
+        # Clear important styles when closing span
+        if tag == "span":
+            self.pending_important_styles = None
+
+        # Clear pending div style when closing a div
+        if tag == "div":
+            self.pending_div_style = None
+
+        # Clear pending inline styles when closing paragraph elements
+        if tag in ["p", "pre"]:
```
Comment (dfop02) [inline html4docx/h4d.py]: Perhaps you miss the h1-h9 here?

**Review by dfop02 — APPROVED** [review]

**Comment by dfop02** [comment]
Awesome @raithedavion ! Merged!

I have one more planned task for this release that I'll probably work on this weekend, so I'm planning to release the `1.1.2` until next week, just for let you know.
